api/src/main/java/jakarta/persistence/EnumeratedValue.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2008, 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Gavin King      - 3.2
+
+
+package jakarta.persistence;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static jakarta.persistence.EnumType.ORDINAL;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Specifies that an annotated field of a Java {@code enum}
+ * type is the source of database column values for an
+ * {@linkplain Enumerated enumerated} mapping. The annotated
+ * field must be declared {@code final}, and must be of type:
+ * <ul>
+ * <li>{@code byte}, {@code short}, or {@code int} for
+ *     {@link EnumType#ORDINAL}, or
+ * <li>{@link String} for {@link EnumType#STRING}.
+ * </ul>
+ * The annotated field must not be null, and must hold a
+ * distinct value for each value of the enum type.
+ *
+ * <pre>
+ *   Example:
+ *
+ *   enum Status {
+ *       OPEN(0), CLOSED(1), CANCELLED(-1);
+ *
+ *       &#064;EnumeratedValue
+ *       final int intValue;
+ *
+ *       Status(int intValue) {
+ *           this.intValue = intValue;
+ *       }
+ *   }
+ * </pre>
+ *
+ * @see Enumerated
+ * @see EnumType
+ *
+ * @since 3.2
+ */
+@Target({FIELD})
```

Comment:
IMO also a `METHOD` should be allowed, as I think most enums will have such a method anyway, whereas I'm not sure if it will have a field.

Reply:
Well the proposal (and it's still just a proposal) was to say that it has to annotate a final field, so that it's enforced to be a constant value for each instance of the enum. A method can, in principle, return a different value each time it's called.

Reply:
>A method can, in principle, return a different value each time it's called.

Sure, but JPA also supports `Access.PROPERTY` which could suffer from the same problem. In reality, users can always shoot their foot, but trying to handle all such cases doesn't seem practical to me.